functions-and-scope.js

```diff
@@ -1,83 +1,223 @@
+    let arrayBeforeTheDecimalPoint = [];
+
+   for (let i = 0; i < numberOfDecimalPlaces; i++) {
+        arrayAfterTheDecimalPoint[i] = Number(theNumberAfterTheDecimalPoint[i]);
+   }
+
+   for (let i = 0; i < numberOfPlacesBeforeTheDecimalPoint; i++){
+       arrayBeforeTheDecimalPoint[i] = Number(numberBeforeTheDecimalPoint[i]);
+   }
+
+   for (let i = numberOfDecimalPlaces - 1; i > 1; i-- ){
+       if (arrayAfterTheDecimalPoint[i] > 4){
+           arrayAfterTheDecimalPoint[i-1] = arrayAfterTheDecimalPoint[i-1] + 1;
+           arrayAfterTheDecimalPoint[i] = 0;
+           arrayAfterTheDecimalPoint.pop();
+       }
+   }
+
+   if(arrayAfterTheDecimalPoint[1] === 10){
+       arrayAfterTheDecimalPoint[1] = 0;
+       arrayAfterTheDecimalPoint[0] = arrayAfterTheDecimalPoint[0] + 1;
+   }
+
+   if (arrayAfterTheDecimalPoint[0] === 10) {
+       arrayBeforeTheDecimalPoint[arrayBeforeTheDecimalPoint.length - 1] = arrayBeforeTheDecimalPoint[arrayBeforeTheDecimalPoint.length - 1] + 1;
+       for (let i = numberOfPlacesBeforeTheDecimalPoint - 1; i > 0; i--) {
+           if (arrayBeforeTheDecimalPoint[i] === 10) {
+               arrayBeforeTheDecimalPoint[i - 1] = arrayBeforeTheDecimalPoint[i - 1] + 1;
+               arrayBeforeTheDecimalPoint[i] = 0;
+           }
+       }
+        arrayAfterTheDecimalPoint[0] = 0;
+       }
+   arrayAfterTheDecimalPoint[2] = ".";
+
+    let resultString  = arrayBeforeTheDecimalPoint[0].toString();
+   for (let i = 1; i < arrayBeforeTheDecimalPoint.length ; i++) {
+       resultString = resultString + arrayBeforeTheDecimalPoint[i].toString();
+   }
+   resultString = resultString + arrayAfterTheDecimalPoint[2] + arrayAfterTheDecimalPoint[0].toString() + arrayAfterTheDecimalPoint[1].toString();
+   let result = Number(resultString);
+   return(result);
+}
+
+let k =  57.4446;
+console.log("\nOpdracht 2c: ");
+console.log("1 " + Math.round(k*100)/100);
+console.log("2 " + (k).toFixed(2));
+console.log("3 " + roundToTwoDecimal(k));
+console.log("Average: " + roundToTwoDecimal(average));
 
 /* Bonusopdracht: hoogste cijfer */
 
 /* 3a: Script schrijven  */
-// Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag hier geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
+// Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is.
+// Je mag hier geen bestaande methoden voor gebruiken.
+// Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
 // * Hoe kan ik iedere waarde van de array langsgaan?
 // * Op welke conditie moet ik checken?
-// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
+// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet,
+// ik dit ergens kan opslaan?
 // Log het antwoord in de terminal.
 
+let hGrade = grades[0];
```

Comment:
Wat goed dat je de eerste waarde van de array pakt in plaats van `0` bijvoorbeeld 👍